scripts/install.sh

```diff
@@ -0,0 +1,183 @@
+#!/bin/bash
+#
+#  Copyright (c) 2024 Red Hat, Inc.
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  You may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+# Script to handle CLI helm installation for k8s and OCP
+#
+# Requires: oc or kubectl
+
+usage() {
+echo "
+This script simplifies and automates the installation process of Helm charts on Kubernetes (K8s) and OpenShift Container Platform (OCP) clusters.
+It detects whether 'oc' or 'kubectl' is installed and ensures that the user is logged into a cluster.
+The script also attempts to detect the cluster router base and updates the Helm chart configuration accordingly.
+
+Usage:
+  $0 [OPTIONS]
+
+Options:
+  --cli <oc|kubectl>          : Specify the CLI tool to use (overrides auto-detection).
+  --router-base <router-base> : Manually provide the cluster router base if auto-detection fails.
+  --release-name <name>       : Specify a custom release name for the Helm chart.
+  --generate-name             : Generate a name for the Helm release (overrides --release-name).
+  --namespace <namespace>     : Specify the namespace for the Helm release (autodetects if not provided).
+  --help                      : Show this help message and exit.
+
+Examples:
+  $0                               # Auto-detects router base and installs the Helm chart
+  $0 --cli kubectl                 # Uses kubectl even if oc is installed
+  $0 --router-base example.com     # Manually specifies the router base and installs the Helm chart
+  $0 --release-name myrelease      # Installs the Helm chart with the specified release name
+  $0 --generate-name               # Generates a name for the Helm release
+"
+}
+
+# Function to check if a command exists
+command_exists() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+# Check if required files and directories exist
+HELM_CHART_DIR="../charts/backstage"
```

Comment:
Another question: unless I misunderstood it, I thought the goal of this script was to ultimately be included in our product docs, so that users could call it instead of running several commands (against either OCP or K8s).
In this case, as it is right now, it would require people to clone this repo first. Is that what we want?
Should we add the chart repo instead?
Or will this be handled in a separate install script?

Reply:
hmm. good point.

so the prod docs would have:
* clone the repo
* run the script

instead of:
* run
* these
* ten
* individual
* steps

And QE could do the same when installing stuff...